src/api.cpp

```diff
@@ -4304,7 +4318,8 @@ bool cl_image_format_to_vulkan_format(const cl_image_format& clformat,
         success = true;
     }
 
-    if (clformat.image_channel_order == CL_LUMINANCE) {
+    if (clformat.image_channel_order == CL_LUMINANCE ||
+        clformat.image_channel_order == CL_INTENSITY) {
         *components = {VK_COMPONENT_SWIZZLE_R, VK_COMPONENT_SWIZZLE_R,
                        VK_COMPONENT_SWIZZLE_R, VK_COMPONENT_SWIZZLE_A};
```

Comment:
This is where you should handle the component ordering. You'll want to use 4 times `VK_COMPONENT_SWIZZLE_R` for `CL_INTENSITY` so the R component of the format you've mapped it on is replicated.